lib/puppet/type/service.rb

```diff
@@ -165,16 +156,6 @@ def sync
       validate do |value|
         raise Puppet::Error.new(_"The 'logonaccount' parameter is mandatory when setting 'logonpassword'.") unless @resource[:logonaccount]
```

Comment:
I'm not sure `@resource[:logonaccount]` is guaranteed to be set before this parameter's `validate` method is called. Usually validation of related props/params is added to the `validate` method for the entire type, such as how we [validate `source` and `content` in the file type(https://github.com/puppetlabs/puppet/blob/c949de2b6eb00b23424eac0fb4f6eb67090ac9eb/lib/puppet/type/file.rb#L409-L440).